runbotics/runbotics-orchestrator-ui/src/src-app/views/credentials/Credential/EditCredential/EditCredentialForm.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { FC, useState } from 'react';
+
+import { Grid, Typography, TextField } from '@mui/material';
+
+import { useSnackbar } from 'notistack';
+
+import { FrontCredentialDto } from 'runbotics-common';
+
+import CustomDialog from '#src-app/components/CustomDialog';
+import If from '#src-app/components/utils/If';
+import useTranslations from '#src-app/hooks/useTranslations';
+
+import { useDispatch } from '#src-app/store';
+import { credentialsActions } from '#src-app/store/slices/Credentials';
+import { Content, Form } from '#src-app/views/utils/FormDialog.styles';
+
+import { inputErrorMessages, InputErrorType } from './EditCredential.utils';
+
+interface EditCredentialFormProps {
+    open: boolean;
+    credential: FrontCredentialDto;
+    collectionId: string;
+    onClose: () => void;
+}
+
+export const EditCredentialForm: FC<EditCredentialFormProps> = ({ open: isOpen, onClose, credential, collectionId }) => {
+    const { translate } = useTranslations();
+    const dispatch = useDispatch();
+    const { enqueueSnackbar } = useSnackbar();
+
+    const [credentialFormState, setCredentialFormState] = useState({ name: credential.name, description: credential.description });
+    const [isFormValid, setIsFormValid] = useState(true);
+
+    const closeDialog = () => {
+        onClose();
+    };
+    const handleSubmit = () => {
+        if (!isFormValid) {
+            enqueueSnackbar(inputErrorMessages[InputErrorType.NAME_IS_REQUIRED], { variant: 'error' });
+            return;
+        }
+
+        dispatch(credentialsActions.updateCredential({ resourceId: credential.id, payload: credentialFormState }))
+            .unwrap()
+            .then(() => {
+                enqueueSnackbar(translate('Credential.Form.Edit.Success', { name: credentialFormState.name }), { variant: 'success' });
+                collectionId
+                    ? dispatch(credentialsActions.fetchAllCredentialsInCollection({ resourceId: `${collectionId}/credentials/` }))
+                    : dispatch(credentialsActions.fetchAllCredentialsAccessibleInTenant());
```

Comment:
this syntax is preferred for returning values, here the best fit is:

if (collectionId) {
dispatch1
} else {
dispatch2 
}